### DISCUSSION TITLE
Stylesheet Info While Building MEI Using Oxygen and Command Line

### DISCUSSION THREAD

**Discussion by yinanazhou** [music-encoding/music-encoding#829] [Q&A] [answered]
Do the command line instructions for building the MEI guidelines use the same stylesheet as the instructions in Oxygen? Are they both using `music-encoding/utils/guidelines_xslt/odd2html.xsl` ? or is the command line method using something in `music-encoding/submodules/Stylesheets` ?

**Comment by ahankinson** [comment, marked as answer]
This would suggest that it uses the same one?

https://github.com/music-encoding/music-encoding/blob/develop/build.xml#L158

**Reply by yinanazhou** [reply]
But if they use the same one, why do all the images not show when using Oxygen, yet when using command lines, it all works well? ([Issue #830](https://github.com/music-encoding/music-encoding/issues/830))

**Reply by ahankinson** [reply]
What are the paths for the images in the compiled HTML?

**Reply by yinanazhou** [reply]
The path template is `../images/modules/neumes/***.png`, e.g.`../images/modules/neumes/NEUME-con.png`. And we have uploaded the missing images to folder `../images/modules/neumes/`. In theory, all the images should have shown as they do when using the command line.

**Reply by ahankinson** [reply]
You're saying that you are getting different output depending on how you compile it.

I am asking what the differences are. What are the paths to the images in the version of the guidelines that work, and what are the paths in the version where they are broken?

**Reply by yinanazhou** [reply]
We are building the same HTML file. The only difference is the method. We are getting different outputs using different methods.

**Reply by yinanazhou** [reply]
I just checked the generated HTML files for both methods, the paths are the same

**Reply by ahankinson** [reply]
I understand that you are generating the same HTML file using two different methods.

I am asking: How does the version of it that works differ from the version of it that does not?

**Reply by ahankinson** [reply]
If the paths are the same, then the images should load in both. Could you please show a screenshot or copy and paste the code here so I can see it? It would be a lot easier to help you that way.

**Reply by yinanazhou** [reply]
Thank you!!! I've attached both.
[generated_html.zip](https://github.com/music-encoding/music-encoding/files/6695772/generated_html.zip)

**Reply by ahankinson** [reply]
The images are given using relative paths: `../images/modules/neumes/NC-rhombusNEW.png`

Are you opening both files in the same directory? If you are in the directory where the HTML file is and type:

`ls ../images/modules/neumes/NC-rhombusNEW.png`

Do you see a file, or do you get 'file not found'?

**Reply by yinanazhou** [reply]
Ohhhhhh! That's the bug! I got it! Thank you soooo much!
We did get that error!

**Reply by ahankinson** [reply]
What's the bug? Could you describe it so that others who might have the same problem will know how to fix it?

**Reply by yinanazhou** [reply]
Oh, sorry! The relative paths are what make the images not showing up when using Oxygen. When switching between different methods, the image paths should be adjusted accordingly.

**Reply by ahankinson** [reply]
Or you could write the output from Oxygen to a different place, presumably?

**Reply by yinanazhou** [reply]
Yes, if the output paths are the same, the relative path will work in both ways. I will try to change the output path from Oxygen.

**Comment by bwbohl** [comment]
Is there something missing from the build description for oXygen that would avoid the problem?

**Comment by yinanazhou** [comment]
Sorry, I was mixing things up. The **relative** path shouldn't be the problem. I checked it again with @martha-thomae today, and we found that when using Oxygen, the `image` folder was not copied into `music-encoding\source\web`. That is why all the images do not show up. After pasting the `image` folder manually into `music-encoding\source\web`, all the images do show up.

**Reply by martha-thomae** [reply]
Hi @bwbohl! So, the thing is:

When you use the command line instructions, you have all these files generated in `music-encoding/dist/guidelines/dev/web`:

![Screen Shot 2021-06-23 at 8 44 15 AM](https://user-images.githubusercontent.com/13948831/123103803-57078380-d404-11eb-91d1-25c5d2ae91ea.png)

You can see that it includes an `images` folder (which is probably copied from the `music-encoding/source/images/` folder).

On the other hand, when you use the Oxygen building instructions, this is the file structure that you get in `music-encoding/source/web`:

![Screen Shot 2021-06-23 at 8 40 21 AM](https://user-images.githubusercontent.com/13948831/123103645-37705b00-d404-11eb-90ab-5bba2053b420.png)

It is missing the `images` folder. Therefore, the relative path to any image (`../images/modules/<something>`), which works perfectly on the folder generated by the command line instructions (first screenshot), doesn't work on the HTML file generated by Oxygen because there is no `images` folder in the first place (second screenshot).

(Thanks for the screenshots, @yinanazhou)

**Reply by bwbohl** [reply]
many thanks for the info

**Reply by musicEnfanthen** [reply]
Follow up question: Why is the command line building to `music-encoding/dist/guidelines/dev/web`, but Oxygen building to `music-encoding/source/web`?

**Comment by bwbohl** [comment]
Ok, apparently after checking music-encoding/build.xml the images folder is not being copied to destination by the XSLT but by Ant. Nevertheless you can run Ant based Transformation Scenarios from within oxygen XML Editor. We should update the build instructions for oXygen accordingly.

**Reply by bwbohl** [reply]
I hope to give it a  try next week, but if anyone is faster, feel free to ;-)

**Reply by o-sapov** [reply]
The problem still persists. The `assests/` folder is almost empty. 
At the very least, the instructions should mention that the 'ant build-guidelines-html' (or better, 'ant dist') task must be run at least once before the XSLT transformation in order to get the assets. 
I can update the `BUILD_OXYGEN.md` file.
